packages/protocol/contracts/common/OwnerUUPSUpgradable.sol

```diff
@@ -58,15 +58,15 @@ abstract contract OwnerUUPSUpgradable is UUPSUpgradeable, OwnableUpgradeable {
     }
 
     function pause() public virtual whenNotPaused {
-        _authorizePause(msg.sender);
         _paused = _TRUE;
         emit Paused(msg.sender);
+        _authorizePause(msg.sender);
```

Comment:
not sure why this will get rid of the new compiler error.

Reply:
It's because in `SignalService.sol` and `AddressManager.sol`, `_authorizePause` always simply reverts no matter what, so any code that follows it is indeed unreachable in those cases.

Reply:
ah, so compiling the sub-contracts will end up giving warnings for the parent-contract. Interesting, but if we think about SignalService and AddressManager, some code unreachable seems to be the right behavior.